docs/en/cloud/manage/cmek.md

```diff
@@ -0,0 +1,97 @@
+
+```json
+ {
+            "Sid": "Allow ClickHouse Access",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::576599896960:role/prod-kms-request-role"
+            },
+            "Action": ["kms:GetPublicKey",
+                    "kms:Decrypt",
+                    "kms:GenerateDataKeyPair",
+                    "kms:Encrypt",
+                    "kms:GetKeyRotationStatus",
+                    "kms:GenerateDataKey",
+                    "kms:DescribeKey"],
+            "Resource": "*"
+        }
+
+```
+![Encryption Key Policy](@site/docs/en/_snippets/images/cmek1.png)
+
+9. Click Finish.
+10. Click the alias of the key you just created.
+11. Use the copy button to copy the ARN.
+
+## Option 2. Configure or Create a KMS key using a CloudFormation stack
+
+ClickHouse provides a simple Cloud Formation stack to deploy the AWS Policy for your key. This method supports both existing KMS keys and creation of new KMS keys for ClickHouse Cloud integration.
+
+**Use an existing KMS key**
+
+1. Login to your AWS account.
+2. Visit [this link](https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/quickcreate?templateURL=https://s3.us-east-2.amazonaws.com/clickhouse-public-resources.clickhouse.cloud/cf-templates/cmek.yaml&stackName=ClickHouseBYOK&param_KMSCreate=false&param_ClickHouseRole=arn:aws:iam::576599896960:role/prod-kms-request-role) to prepare the CloudFormation template.
+3. Enter the ARNs of the KMS key(s) you want to use (comma separated with no spaces in between).
+4. Accept "I acknowledge that AWS CloudFormation might create IAM resources with custom names." and click __Create stack__.
+5. Make note of the `RoleArn` and the `KeyArn` in the stack output as you will need these for the next step.
+
+
+**Create a new KMS key**
+
+1. Login to your AWS account.
+2. Visit [this link](https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/quickcreate?templateURL=https://s3.us-east-2.amazonaws.com/clickhouse-public-resources.clickhouse.cloud/cf-templates/cmek.yaml&stackName=ClickHouseBYOK&param_KMSCreate=true&param_ClickHouseRole=arn:aws:iam::576599896960:role/prod-kms-request-role) to prepare the CloudFormation template.
+4. Accept "I acknowledge that AWS CloudFormation might create IAM resources with custom names." and click __Create stack__.
+5. Make note of the `KeyArn` in the stack output as you will need this for the next step.
+
+## Option 3. Create a KMS key via Terraform
+
+For advanced users who want to deploy the key via terraform, check out the AWS provider documentation [here](https://registry.terraform.io/providers/hashicorp/aws/3.26.0/docs/resources/kms_key).
+
+# Step 2. Starting a Service with Customer Managed Encryption Keys
+
+1. Log into your ClickHouse Cloud account.
+2. Go to the Services screen if you are not already there.
+3. Click __New Service__.
+4. Select your Cloud provider, Region and name your service.
```

Comment:
Will this work on Dev services, or do we  require Production?

Reply:
It is only available for production resources in AWS.

Reply:
It also appears to be sitting behind a feature flag. Confirming.

Reply:
Confirmed and updated the notice at the top of the doc.